### PR TITLE
feat(pre-req): enable manipulation of proxy and certificate through insomnia.request - INS-3379

### PR DESCRIPTION
**Pull request by ihexxa**
### Please review this PR (https://github.com/Kong/insomnia/pull/7170) firstly as current one is based on it
This is change is not part of request manipulation PR because this basically manipulate values from settings and clientCertificates, and for making PRs small.

Changes:
- [x] enabled manipulation certificate and proxy through `insomnia.request`
- [x] added smoke tests
- [x] added UTs

### How to test
Run this script and check if it runs well:

```javascript
// print current proxy and modify it
console.log(insomnia.request.proxy.getProxyUrl());
insomnia.request.proxy.update({
        host: 'localhost',
        match: '<all_urls>',
        port: 8888,
        tunnel: false,
        authenticate: false,
        username: '',
        password: '',
});

// print current certificate and modify it
// print the original one
console.log('key:', insomnia.request.certificate.key.src);
console.log('cert:', insomnia.request.certificate.cert.src);
console.log('passphrass:', insomnia.request.certificate.passphrass);
console.log('pfx:', insomnia.request.certificate.pfx.src);
// update
insomnia.request.certificate.update({
    disabled: true,
    key: {src: 'invalid.key'},
    cert: {src: 'invalid.cert'},
    passphrase: '',
    pfx: {src: ''},
});
```

### CODE REVIEW

File: packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts [hunk 1]
```diff
@@ -405,4 +407,97 @@ test.describe('pre-request features tests', async () => {
...
+                    tunnel: false,
+                    authenticate: false,
+                    username: '',
+                    password: '',
+            });
+        `);
+
+        // TODO: wait for body and pre-request script are persisted to the disk
+        // should improve this part, we should avoid sync this state through db as it introduces race condition
+        await page.waitForTimeout(500);
```
Comment (jackkav) [inline packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts:231]: This kind of test is much easier to maintain than the abstract tests in this file. However there is always a workaround to ensure the data has been written to disk besides using a wait. Waits should be avoided at all costs in e2e tests. I would suggest trying to click something or navigate somewhere instead.

File: packages/insomnia/src/sdk/objects/auth.ts [hunk 2]
```diff
@@ -335,14 +335,15 @@ export class RequestAuth extends Property {
...
+
     const findValueInKvArray = (targetKey: string, kvs?: { key: string; value: string }[]) =>
         kvs?.find(({ key }) => key === targetKey)?.value || '';
 
     const findValueInOauth2Options = (
         targetKey: string,
         kvs?: { key: string; value: string }[] | OAuth2AuthOption[]
     ) => {
-        if (!kvs) {
+        if (kvs == null) {
```
Comment (jackkav) [inline packages/insomnia/src/sdk/objects/auth.ts]: Loose null checks are a smell

File: packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts (hunk 1 above)
Comment (ihexxa) [reply packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts:231]: Good point, as there are almost 10 test cases which shares the operation workflow, turning it into normal test will introduce 10x duplications, so probably we should be careful of this. Regarding the waiting, I am also struggling on this, it requires to wait a relatively stable period or this test would be flaky, is it possible to observe some events which are stable?

File: packages/insomnia/src/sdk/objects/auth.ts (hunk 2 above)
Comment (ihexxa) [reply packages/insomnia/src/sdk/objects/auth.ts]: It slips into this change in merging, updated.

File: packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts (hunk 1 above)
Comment (ihexxa) [reply packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts:231]: Most of tests are updated to use predefined request.

**Review by jackkav — APPROVED** [review]

**Pull request merged**